src/frontend/javascript/cookie-functions.js

```diff
@@ -0,0 +1,186 @@
+/* istanbul ignore file */
+
+function Cookies () {
+    this.cookieName = 'govuk-paas-cookie-policy'
+    this.cookieDomain = 'cloud.service.gov.uk'
+    this.cookieDuration = 365
+    this.trackingId = 'UA-43115970-5'
+    // disable tracking by default
+    window['ga-disable-' + this.trackingId] = true;
+}
+
+Cookies.prototype.hasConsentForAnalytics = function () {
+  var consentCookie = JSON.parse(this.getCookie(this.cookieName));
+  return consentCookie ? consentCookie.analytics : false
+}
+
+Cookies.prototype.initAnalytics = function() {
+
+  // guard against being called more than once
+  if (!('GoogleAnalyticsObject' in window)) {
+
+    window['ga-disable-' + this.trackingId] = false;
+
+    // Load GTM
+    this.loadGtmScript()
+    this.setupGtm()
+  }
+}
+
+Cookies.prototype.initCookieBanner = function ($module) {
+  this.$module = $module
+
+  this.$module.hideCookieMessage = this.hideCookieMessage.bind(this)
+  this.$module.showBannerConfirmationMessage = this.showBannerConfirmationMessage.bind(this)
+  this.$module.setBannerCookieConsent = this.setBannerCookieConsent.bind(this)
+  this.$module.cookieBannerConfirmationMessage = this.$module.querySelector('.cookie-banner__confirmation')
+  
+  /* test already test this function */
+
+  if (!this.$module) {
```

Comment:
What is this check trying to achieve? It seems out of place given we already set a bunch of properties on the lines above.

Reply:
if there is no dom element don't run the rest